werkzeug/serving.py

```diff
@@ -209,6 +210,14 @@ def execute(app):
             execute(self.server.app)
         except (socket.error, socket.timeout) as e:
             self.connection_dropped(e, environ)
+        except IOError, e:
+            if e.errno == errno.EPIPE:
+                self.server.log('error', 'IOError, Broken Pipe: possibly a broken connection.')
+            else:
```

Comment:
This doesn't work as expected you'd need to replicate the whole logic from `except Exception` (you could get an EPIPE from application code too, in which case you do want to set a proper 500 reply). Since repeating the code would be extremely ugly I'd rather nest try/except blocks. Something like this:

```python
try:
    try:
        foo()
    except IOError as e:
        if e.errno != errno.EPIPE:
            raise
        # handle EPIPE here
except (socket.error, socket.timeout) as e:
    ...
except Exception:
    ...
```